uaclient/lock.py

```diff
@@ -0,0 +1,89 @@
+import functools
+import logging
+import os
+import time
+
+from uaclient import config, exceptions
+
+LOG = logging.getLogger("ua.lock")
+
+# Set a module-level callable here so we don't have to reinstantiate
+# UAConfig in order to determine dynamic data_path exception handling of
+# main_error_handler
+clear_lock_file = None
+
+
+def clear_lock_file_if_present():
+    global clear_lock_file
+    if clear_lock_file:
+        clear_lock_file()
+
+
+class SingleAttemptLock:
+    """
+    Context manager for gaining exclusive access to the lock file.
+    Create a lock file if absent. The lock file will contain a pid of the
+        running process, and a customer-visible description of the lock holder.
```

Comment:
I think we can remove this extra space here

Reply:
Yep my bad